components/script/script_runtime.rs

```diff
@@ -826,7 +832,46 @@ impl StreamConsumer {
 }
 
 #[allow(unsafe_code)]
-unsafe extern "C" fn report_stream_error_callback(_cx: *mut JSContext, error_code: usize) {
+unsafe extern "C" fn ConsumeStreamCallback(
+    _cx: *mut RawJSContext,
+    obj: HandleObject,
+    _mimeType: MimeType,
+    _consumer: *mut JSStreamConsumer,
+) {
+    let cx = JSContext::from_ptr(_cx);
+    let global = GlobalScope::from_context(*cx);
+    let returnValue = Promise::new(&global);
+    let mut handleobj = UndefinedValue();
+    GetProxyPrivate(*obj, &mut handleobj);
+    rooted!(in(*cx) let target = handleobj.to_object());
```

Comment:
It's not clear to me why these lines are necessary. Why can't we use `obj` as the argument to root_from_handleobject?

Reply:
The consume_stream function takes a JS::HandleObject as an argument while the root_from_handle_obj expects a mozjs::rust::HandleObject.

Reply:
Ah, you can use `mozjs::rust::HandleObject::from_raw(obj)` in that case.

Reply:
Done